### PR TITLE
Expansion Support

### PR DESCRIPTION
**Pull request by prankulmahajan**
Dont merge till https://github.com/IBM/ibmcloud-volume-file-vpc/pull/10/files is closed

### CODE REVIEW

**Comment by prankulmahajan** [comment]
Tested the functionality and is working as expected.
- [x] IKS Prod
```
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % ibmcloud ks cluster config -c mycluster-us-south-1-cx2.2x4
OK
The configuration for mycluster-us-south-1-cx2.2x4 was downloaded successfully.

Added context for mycluster-us-south-1-cx2.2x4 to the current kubeconfig file.
You can now execute 'kubectl' commands against your cluster. For example, run 'kubectl get nodes'.
If you are accessing the cluster for the first time, 'kubectl' commands might fail for a few seconds while RBAC synchronizes.
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods -n kube-system -o wide | grep file 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % ./deploy/kubernetes/driver/kubernetes/deploy-vpc-file-driver.sh stage
This will install 'stage' version of vpc csi driver!
serviceaccount/ibm-vpc-file-controller-sa created
serviceaccount/ibm-vpc-file-node-sa created
clusterrole.rbac.authorization.k8s.io/vpc-file-driver-registrar-role created
clusterrole.rbac.authorization.k8s.io/vpc-file-external-resizer-role created
clusterrole.rbac.authorization.k8s.io/vpc-file-provisioner-role created
clusterrolebinding.rbac.authorization.k8s.io/vpc-file-driver-registrar-binding created
clusterrolebinding.rbac.authorization.k8s.io/vpc-file-external-resizer-binding created
clusterrolebinding.rbac.authorization.k8s.io/vpc-file-provisioner-binding created
configmap/ibm-vpc-file-csi-configmap created
statefulset.apps/ibm-vpc-file-csi-controller created
daemonset.apps/ibm-vpc-file-csi-node created
csidriver.storage.k8s.io/vpc.file.csi.ibm.io created
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods -n kube-system -o wide | grep file                  
ibm-vpc-file-csi-controller-0                         0/5     ContainerCreating   0          13s    <none>         10.240.0.30   <none>           <none>
ibm-vpc-file-csi-node-99qkm                           0/4     ContainerCreating   0          12s    <none>         10.240.0.30   <none>           <none>
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods -n kube-system -o wide | grep file
ibm-vpc-file-csi-controller-0                         0/5     ContainerCreating   0          19s    <none>         10.240.0.30   <none>           <none>
ibm-vpc-file-csi-node-99qkm                           0/4     ContainerCreating   0          18s    <none>         10.240.0.30   <none>           <none>
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods -n kube-system -o wide | grep file
ibm-vpc-file-csi-controller-0                         5/5     Running            0          30s    172.22.60.50   10.240.0.30   <none>           <none>
ibm-vpc-file-csi-node-99qkm                           4/4     Running            0          29s    172.22.60.51   10.240.0.30   <none>           <none>
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc
No resources found in default namespace.
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl apply -f local_logs/5iops-file-expansion.yaml 
storageclass.storage.k8s.io/5iops-file-expansion created
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl apply -f examples/kubernetes/validPVC.yaml 
persistentvolumeclaim/csi-file-expansion-4 created
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % watch kubectl get pvc
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl apply -f examples/kubernetes/validPOD.yaml 
deployment.apps/testpod-expansion4 created


prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc
NAME                   STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
csi-file-expansion-4   Bound    pvc-6a493458-9f1e-4bc7-b278-d4aae54e1874   20Gi       RWX            5iops-file-expansion   5h25m

prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl describe pvc csi-file-expansion-4
Name:          csi-file-expansion-4
Namespace:     default
StorageClass:  5iops-file-expansion
Status:        Bound
Volume:        pvc-6a493458-9f1e-4bc7-b278-d4aae54e1874
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: vpc.file.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      20Gi
Access Modes:  RWX
VolumeMode:    Filesystem
Used By:       testpod-expansion4-7656585c6c-7jc2t
               testpod-expansion4-7656585c6c-bnm9b
Events:
  Type     Reason                      Age                   From                                                                                    Message
  ----     ------                      ----                  ----                                                                                    -------
  Normal   Provisioning                2m27s                 vpc.file.csi.ibm.io_ibm-vpc-file-csi-controller-0_cf2a824b-2c43-4106-bccf-32b313adc2b8  External provisioner is provisioning volume for claim "default/csi-file-expansion-4"
  Normal   ExternalProvisioning        2m7s (x3 over 2m27s)  persistentvolume-controller                                                             waiting for a volume to be created, either by external provisioner "vpc.file.csi.ibm.io" or manually created by system administrator
  Normal   ProvisioningSucceeded       2m2s                  vpc.file.csi.ibm.io_ibm-vpc-file-csi-controller-0_cf2a824b-2c43-4106-bccf-32b313adc2b8  Successfully provisioned volume pvc-6a493458-9f1e-4bc7-b278-d4aae54e1874
  Normal   Resizing                    60s                   external-resizer vpc.file.csi.ibm.io                                                    External resizer is resizing volume pvc-6a493458-9f1e-4bc7-b278-d4aae54e1874
  Warning  ExternalExpanding           60s                   volume_expand                                                                           Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
  Normal   FileSystemResizeRequired    49s                   external-resizer vpc.file.csi.ibm.io                                                    Require file system resize of volume on node
  Normal   FileSystemResizeSuccessful  12s                   kubelet                                                                                 MountVolume.NodeExpandVolume succeeded for volume "pvc-6a493458-9f1e-4bc7-b278-d4aae54e1874"
  
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods
NAME                                  READY   STATUS    RESTARTS   AGE
testpod-expansion4-7656585c6c-7jc2t   1/1     Running   0          5h25m
testpod-expansion4-7656585c6c-bnm9b   1/1     Running   0          5h25m


prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec testpod-expansion4-7656585c6c-7jc2t -it -- bash
root@testpod-expansion4-7656585c6c-7jc2t:/# df -h
Filesystem                                                                                    Size  Used Avail Use% Mounted on
overlay                                                                                        99G   16G   78G  18% /
tmpfs                                                                                          64M     0   64M   0% /dev
tmpfs                                                                                         2.0G     0  2.0G   0% /sys/fs/cgroup
fsf-dal1051b-fz.adn.networklayer.com:/nxg_s_voll_mz0717_08536c92_5c70_48cf_8953_699ff7f4f8a1   20G  384K   20G   1% /myvol1
/dev/vda2                                                                                      99G   16G   78G  18% /etc/hosts
shm                                                                                            64M     0   64M   0% /dev/shm
tmpfs                                                                                         2.0G   16K  2.0G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs                                                                                         2.0G     0  2.0G   0% /proc/acpi
tmpfs                                                                                         2.0G     0  2.0G   0% /proc/scsi
tmpfs                                                                                         2.0G     0  2.0G   0% /sys/firmware
```

**Comment by prankulmahajan** [comment]
- [x] IKS Stage
```
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods -n kube-system -o wide | grep file
ibm-vpc-file-csi-controller-0                         5/5     Running   0              41h   172.23.52.81   10.240.0.29   <none>           <none>
ibm-vpc-file-csi-node-r6q5s                           4/4     Running   0              41h   172.23.52.82   10.240.0.29   <none>           <none>
ibm-vpc-file-csi-node-tbqv2                           4/4     Running   0              41h   172.23.60.11   10.240.0.30   <none>           <none>
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl apply -f examples/kubernetes/validPVC.yaml
persistentvolumeclaim/csi-file-expansion-4 created
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % watch kubectl get pvc                              
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc
NAME                   STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
csi-file-expansion-4   Bound    pvc-2e0d8cf6-a9e3-488e-93a4-b630dfcd6e39   10Gi       RWX            5iops-file-expansion   31s
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl apply -f examples/kubernetes/validPOD.yaml
deployment.apps/testpod-expansion4 created
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % watch kubectl get pods
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods
NAME                                  READY   STATUS    RESTARTS   AGE
testpod-expansion4-64df5b79cc-jxwhr   1/1     Running   0          23s
testpod-expansion4-64df5b79cc-w5wlv   1/1     Running   0          23s
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl edit pvc csi-file-expansion-4 
^[[Bpersistentvolumeclaim/csi-file-expansion-4 edited
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc                                   
NAME                   STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
csi-file-expansion-4   Bound    pvc-2e0d8cf6-a9e3-488e-93a4-b630dfcd6e39   10Gi       RWX            5iops-file-expansion   82s
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl describe pvc csi-file-expansion-4   
Name:          csi-file-expansion-4
Namespace:     default
StorageClass:  5iops-file-expansion
Status:        Bound
Volume:        pvc-2e0d8cf6-a9e3-488e-93a4-b630dfcd6e39
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: vpc.file.csi.ibm.io
               volume.kubernetes.io/storage-provisioner: vpc.file.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      10Gi
Access Modes:  RWX
VolumeMode:    Filesystem
Used By:       testpod-expansion4-64df5b79cc-jxwhr
               testpod-expansion4-64df5b79cc-w5wlv
Conditions:
  Type                      Status  LastProbeTime                     LastTransitionTime                Reason  Message
  ----                      ------  -----------------                 ------------------                ------  -------
  FileSystemResizePending   True    Mon, 01 Jan 0001 00:00:00 +0000   Fri, 07 Jan 2022 17:01:46 +0530           Waiting for user to (re-)start a pod to finish file system resize of volume on node.
Events:
  Type     Reason                    Age                From                                                                                    Message
  ----     ------                    ----               ----                                                                                    -------
  Normal   Provisioning              94s                vpc.file.csi.ibm.io_ibm-vpc-file-csi-controller-0_33b550d2-692b-4e7b-b8e6-f22c96e8db55  External provisioner is provisioning volume for claim "default/csi-file-expansion-4"
  Normal   ExternalProvisioning      78s (x3 over 94s)  persistentvolume-controller                                                             waiting for a volume to be created, either by external provisioner "vpc.file.csi.ibm.io" or manually created by system administrator
  Normal   ProvisioningSucceeded     70s                vpc.file.csi.ibm.io_ibm-vpc-file-csi-controller-0_33b550d2-692b-4e7b-b8e6-f22c96e8db55  Successfully provisioned volume pvc-2e0d8cf6-a9e3-488e-93a4-b630dfcd6e39
  Warning  ExternalExpanding         18s                volume_expand                                                                           Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
  Normal   Resizing                  18s                external-resizer vpc.file.csi.ibm.io                                                    External resizer is resizing volume pvc-2e0d8cf6-a9e3-488e-93a4-b630dfcd6e39
  Normal   FileSystemResizeRequired  7s                 external-resizer vpc.file.csi.ibm.io                                                    Require file system resize of volume on node
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % watch kubectl get pvc                             
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc
NAME                   STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
csi-file-expansion-4   Bound    pvc-2e0d8cf6-a9e3-488e-93a4-b630dfcd6e39   20Gi       RWX            5iops-file-expansion   2m14s
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec testpod-expansion4-64df5b79cc-jxwhr -it -- bash
root@testpod-expansion4-64df5b79cc-jxwhr:/# df -h
Filesystem                                                                                             Size  Used Avail Use% Mounted on
overlay                                                                                                 99G   13G   82G  14% /
tmpfs                                                                                                   64M     0   64M   0% /dev
tmpfs                                                                                                  2.0G     0  2.0G   0% /sys/fs/cgroup
fsf-dal1097a-vs90-stg-fz.adn.networklayer.com:/nxg_s_voll_mz0716_d574992f_6bd0_42b9_9f65_01f773a629cd   20G  320K   20G   1% /myvol1
/dev/vda2                                                                                               99G   13G   82G  14% /etc/hosts
shm                                                                                                     64M     0   64M   0% /dev/shm
tmpfs                                                                                                  2.8G   16K  2.8G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs                                                                                                  2.0G     0  2.0G   0% /proc/acpi
tmpfs                                                                                                  2.0G     0  2.0G   0% /proc/scsi
tmpfs                                                                                                  2.0G     0  2.0G   0% /sys/firmware
root@testpod-expansion4-64df5b79cc-jxwhr:/# exit
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl delete deploy testpod-expansion4 
deployment.apps "testpod-expansion4" deleted
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % watch kubectl get pods                                      
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods
No resources found in default namespace.
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl delete pvc csi-file-expansion-4 
persistentvolumeclaim "csi-file-expansion-4" deleted
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % watch kubectl get pvc                  
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % 
prankul@Prankuls-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc
No resources found in default namespace.
```

File: deploy/kubernetes/driver/kubernetes/manifests/controller-server.yaml [hunk 1]
```diff
@@ -46,6 +46,29 @@ spec:
...
+          args:
+            - "--v=5"
+            - "--csi-address=$(ADDRESS)"
+            - "--timeout=600s"
+          env:
+            - name: ADDRESS
+              value: /csi/csi-vpc-file.sock
+          resources:
+            limits:
+              cpu: 100m
```
Comment (sameshai) [inline deploy/kubernetes/driver/kubernetes/manifests/controller-server.yaml:64]: Did we confirm with block resizer and we are using optimal resources

**Review by sameshai — CHANGES_REQUESTED** [review]
@prankulmahajan Can you also check the sanity part. Also can you check existing coverage and new coverage. Also we need to make changes for internal repo

File: deploy/kubernetes/driver/kubernetes/manifests/controller-server.yaml (hunk 1 above)
Comment (prankulmahajan) [reply deploy/kubernetes/driver/kubernetes/manifests/controller-server.yaml:64]: Added same resources as block

Comment (prankulmahajan) [reply deploy/kubernetes/driver/kubernetes/manifests/controller-server.yaml:64]: Test

**Comment by prankulmahajan** [comment]
Tested scenario : "Writing data into the attached fileshare and then expanding"

```
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl get pods -n kube-system | grep file                                                      
ibm-vpc-file-csi-controller-0                         5/5     Running   0          26h
ibm-vpc-file-csi-node-4l9k5                           4/4     Running   0          26h
ibm-vpc-file-csi-node-78mb6                           4/4     Running   0          26h
ibm-vpc-file-csi-node-hcs65                           4/4     Running   0          26h
ibm-vpc-file-csi-node-jwlnv                           4/4     Running   0          26h
ibm-vpc-file-csi-node-q4fld                           4/4     Running   0          26h
ibm-vpc-file-csi-node-vrrsb                           4/4     Running   0          26h

➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl apply -f local_logs/5iops-file-expansion.yaml 
storageclass.storage.k8s.io/5iops-file-expansion created
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl get sc | grep expansion                      
5iops-file-expansion                          vpc.file.csi.ibm.io    Delete          Immediate              true                   13s

➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl apply -f examples/kubernetes/validPVC.yaml 
persistentvolumeclaim/csi-file-pvc-good created
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ watch kubectl get pvc
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl get pvc
NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
csi-file-pvc-good   Bound    pvc-abd1bf94-af62-49dc-be07-8f218f13045d   10Gi       RWX            5iops-file-expansion   34s
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl apply -f examples/kubernetes/validPOD.yaml 
deployment.apps/testpodrwm created
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ watch kubectl get pods
                                                                                                                                                    
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl get pvc
^C
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl get pods
NAME                          READY   STATUS    RESTARTS   AGE
testpodrwm-67cd77b44c-5nbxs   1/1     Running   0          28s
testpodrwm-67cd77b44c-z9h6t   1/1     Running   0          28s
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl exec testpodrwm-67cd77b44c-5nbxs -it -- bash
root@testpodrwm-67cd77b44c-5nbxs:/# df -h | grep tmp
tmpfs                                                                                                   64M     0   64M   0% /dev
tmpfs                                                                                                  3.9G     0  3.9G   0% /sys/fs/cgroup
tmpfs                                                                                                  3.9G   16K  3.9G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs                                                                                                  3.9G     0  3.9G   0% /proc/acpi
tmpfs                                                                                                  3.9G     0  3.9G   0% /proc/scsi
tmpfs                                                                                                  3.9G     0  3.9G   0% /sys/firmware
root@testpodrwm-67cd77b44c-5nbxs:/# 
root@testpodrwm-67cd77b44c-5nbxs:/# 
root@testpodrwm-67cd77b44c-5nbxs:/# 
root@testpodrwm-67cd77b44c-5nbxs:/# df -h | grep /myvol
fsf-dal1297a-vs90-stg-fz.adn.networklayer.com:/nxg_s_voll_mz0726_eeeb6418_e6a4_4c81_99fc_dae1325d7735  9.7G  320K  9.6G   1% /myvol
root@testpodrwm-67cd77b44c-5nbxs:/# 
root@testpodrwm-67cd77b44c-5nbxs:/# cd myvol/
root@testpodrwm-67cd77b44c-5nbxs:/myvol# ls
root@testpodrwm-67cd77b44c-5nbxs:/myvol# echo "hi" > sample.txt
root@testpodrwm-67cd77b44c-5nbxs:/myvol# ls
sample.txt
root@testpodrwm-67cd77b44c-5nbxs:/myvol# cat sample.txt 
hi
root@testpodrwm-67cd77b44c-5nbxs:/myvol# echo "hi there" > sample2.txt
root@testpodrwm-67cd77b44c-5nbxs:/myvol# 
root@testpodrwm-67cd77b44c-5nbxs:/myvol# 
root@testpodrwm-67cd77b44c-5nbxs:/myvol# ls
sample.txt  sample2.txt
root@testpodrwm-67cd77b44c-5nbxs:/myvol# exit
exit
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl get pods                                    
NAME                          READY   STATUS    RESTARTS   AGE
testpodrwm-67cd77b44c-5nbxs   1/1     Running   0          2m8s
testpodrwm-67cd77b44c-z9h6t   1/1     Running   0          2m8s
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl edit pvc csi-file-pvc-good 
persistentvolumeclaim/csi-file-pvc-good edited
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ watch kubectl get pvc                               
                                                                                                                                                    
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl get pvc
NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
csi-file-pvc-good   Bound    pvc-abd1bf94-af62-49dc-be07-8f218f13045d   20Gi       RWX            5iops-file-expansion   4m16s
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ 
➜  ibm-vpc-file-csi-driver git:(expansion) ✗ kubectl exec testpodrwm-67cd77b44c-5nbxs -it -- bash
root@testpodrwm-67cd77b44c-5nbxs:/# df -h | grep /myvol
fsf-dal1297a-vs90-stg-fz.adn.networklayer.com:/nxg_s_voll_mz0726_eeeb6418_e6a4_4c81_99fc_dae1325d7735   20G  320K   20G   1% /myvol
root@testpodrwm-67cd77b44c-5nbxs:/# 
root@testpodrwm-67cd77b44c-5nbxs:/# cd myvol/
root@testpodrwm-67cd77b44c-5nbxs:/myvol# ls
sample.txt  sample2.txt
root@testpodrwm-67cd77b44c-5nbxs:/myvol# cat sample
cat: sample: No such file or directory
root@testpodrwm-67cd77b44c-5nbxs:/myvol# cat sample.txt 
hi
root@testpodrwm-67cd77b44c-5nbxs:/myvol# cat sample2.txt 
hi there
root@testpodrwm-67cd77b44c-5nbxs:/myvol# exit
exit

```

**Review by sameshai — APPROVED** [review]
LGTM

**Pull request merged**